drivers/random/random_nrf5.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2017 Exati Tecnologia Ltda.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <random.h>
+#include <atomic.h>
+#include <nrf_rng.h>
+
+struct random_nrf5_dev_data {
+	atomic_t user_count;
+};
+
+#define DEV_DATA(dev) \
+	((struct random_nrf5_dev_data *)(dev)->driver_data)
+
+static inline s16_t random_nrf5_get_u8(void)
+{
+	u8_t value;
+	u32_t timeout = 0;
+
+	while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY)) {
```

Comment:
This loop is not low power, and use system clock interface to measure timeout (nRF51 is 16MHz CM0 CPU and nRF52 is 64MHz CM4F CPU).

Reply:
I understand that the loop could be low power if I used the __WFE, __SVE, __WFE combo, but can I still have a timeout using these constructs? I don't understand the exact mechanics of it. Apart from "one-time" events like VALRDY, does the system generate events at fixed intervals, letting the loop continue?

When you say system clock interface, do you mean `k_cycle_get_32()` and related functions or something else?